### ISSUE TITLE
Add a flag to generate a directory if non exists

### ISSUE THREAD

**Issue by Aghassi** [shelljs/shelljs#827]
https://github.com/shelljs/shelljs/blob/9035b27403c7e3c3dc801939d20dec8fe6143c06/src/cp.js#L240

With regard to the line above, the current feature set `cp` will throw an error. I find that I just end up wrapping this method to check for the dir anyway and then doing a `mkdir` call. It would be nice to have a flag to allow users to request the missing folder be generated on their behalf. I have no problem writing this feature but first wanted to request it in case there is a reason that it doesn't exist already.

**Comment by nfischer** [comment]
Does unix `cp` have a similar feature?

As I'm aware, the typical pattern in shell scripts is:

```sh
$ mkdir -p folder/which/may/exist
$ cp file.txt folder/which/may/exist
```

Same thing should translate into ShellJS easily.

**Comment by Aghassi** [comment]
It doesn't exist in a Unix shell. You are right that is the pattern. I was just curious for convenience sake so users wouldn't have to make a helper method to do just that. If it isn't something you guys want to do because you want to be true to Unix, I understand. Was just curious :)

**Comment by nfischer** [comment]
Yeah, we try to stick close to unix features, and the bash pattern isn't too hard to follow. This is a reasonable feature request if we ever add lots of non-posix features.
